src/main/java/com/shopify/store/ShoppingCart.java

```diff
@@ -0,0 +1,55 @@
+package com.shopify.store;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShoppingCart {
+    private List<LineItem> items;
+    private Discount discountZero;
+    private Discount discountBOGO;
+    private Discount discountPercent;
```

Comment:
Scale. If you had 100 different types of discounts, would you have 100 different discount fields here?